scripts/us_census/acs5yr/subject_tables/common/acs_spec_validator.py

```diff
@@ -0,0 +1,849 @@
+                                        column_name)
+                                    ret_dict[prop_token][
+                                        'possibleParents'].extend(
+                                            temp_dict[prop][:j])
+                    # if the token is near the leaf but not used as a specialisation, it potentially has a base value
+                    if j > 0 and temp_flag:
+                        if prop_token not in ret_dict:
+                            ret_dict[prop_token] = {}
+                            ret_dict[prop_token]['column'] = [column_name]
+                            ret_dict[prop_token]['possibleParents'] = temp_dict[
+                                prop][:j]
+                        else:
+                            ret_dict[prop_token]['column'].append(column_name)
+                            ret_dict[prop_token]['possibleParents'].extend(
+                                temp_dict[prop][:j])
+
+    for prop_token in ret_dict:
+        ret_dict[prop_token]['possibleParents'] = list(
+            set(ret_dict[prop_token]['possibleParents']))
+
+    return ret_dict
+
+
+def find_multiple_measurement(column_name_list: list,
+                              spec_dict: dict,
+                              delimiter: str = '!!') -> list:
+    """Check if any column is getting multiple measurement associated.
+
+      Args:
+        column_name_list: List of all columns after dropping ignored columns.
+        spec_dict: Dict obj containing configurations for the import.
+        delimiter: delimiter seperating tokens within single column name string.
+      
+      Returns:
+        List of columns that have multiple measurement associated with it.
+  """
+    ret_list = []
+
+    # tokenList = getTokensListFromColumnList(columnNameList, delimiter)
+    for column_name in column_name_list:
+        if 'measurement' in spec_dict:
+            temp_flag = False
+            for token in column_name.split(delimiter):
+                if token in spec_dict['measurement']:
+                    if temp_flag:
+                        ret_list.append(column_name)
+                    temp_flag = True
+    return ret_list
+
+
+def find_multiple_population(column_name_list: list,
+                             spec_dict: dict,
+                             delimiter: str = '!!') -> list:
+    """Check if any column is getting multiple population types associated.
+
+      Args:
+        column_name_list: List of all columns after dropping ignored columns.
+        spec_dict: Dict obj containing configurations for the import.
+        delimiter: delimiter seperating tokens within single column name string.
+      
+      Returns:
+        List of columns that have multiple populationTypes associated with it.
+  """
+    ret_list = []
+
+    # tokenList = getTokensListFromColumnList(columnNameList, delimiter)
+    for column_name in column_name_list:
+        if 'populationType' in spec_dict:
```

Comment:
Can move `if 'populationType' in spec_dict:` outside the loop

Reply:
fixed. Thanks for the suggestion.